sources/web/datalab/auth.ts

```diff
@@ -128,6 +128,17 @@ export function isSignedIn(): boolean {
   return (process.env.DATALAB_ENV != 'local' || fs.existsSync(appCredFile));
 }
 
+function setOauth2Client(parsed_url: any): void {
+  if (!oauth2Client) {
+    var OAuth2:any = google.auth.OAuth2;
+    // TODO(ojarjur): Ideally, we would get the host and port from the parsed_url
+    // rather than hard-coding them. However, the client ID and secret we are using
+    // are limited to localhost only. We should consider making this configurable, or
+    // using the OAuth flow for non-web applications.
+    oauth2Client = new OAuth2(clientId, clientSecret, 'http://localhost:8081/oauthcallback');
```

Comment:
I see. We do plan to use the same port for docker and server.

But I am not sure whether we can get the port from the client request (in this case, 8081) from our web server. Can we?